### ISSUE TITLE
Type inference incorrectly propagates props from targeted styled components

### ISSUE THREAD

**Issue by rvetere** [jantimon/next-yak#180]
When using a component that targets another component using the nested selector syntax, TypeScript incorrectly propagates the prop types from the targeted component to the parent component.
 
## Reproduction
 
```tsx
const Icon = styled.svg<{ $active: boolean }>`
  /* icon styles */
`;
 
const Button = styled.button`
  ${Icon} {
    color: red;
  }
`;
 
// TypeScript incorrectly requires $active prop here
<Button $active={true}>Click me</Button> // Should not require $active
```
 
## Expected Behavior
 
The `Button` component should not inherit the `$active` prop type from `Icon` since it's only using `Icon` as a selector for styling nested elements. The `$active` prop is only meaningful when used directly on the `Icon` component.
 
## Actual Behavior
 
TypeScript enforces the `$active` prop on the `Button` component even though:
1. The prop is never used in Button's styles
2. The nested selector syntax (`${Icon}`) is only meant for targeting the component in CSS, not for prop inheritance
3. next-yak does not actually do anything with these props at runtime in this context
 
## Impact
 
This incorrect type inference forces users to add unnecessary props to parent components, leading to confusing and incorrect usage patterns.

**Comment by jantimon** [comment]
sounds like a job for @Mad-Kat 😄

**Comment by Mad-Kat** [comment]
Looks like it already works?

[Playground](https://www.typescriptlang.org/play/?jsx=4#code/JYWwDg9gTgLgBAbwM4wJ4BsCmATAvnAMyghDgCIA7TADxgFpUBDAazIG4AoDgYwgpTgBJXhTgBeOCgw4AdEgBuAcwA8COABJG3GMHmYAXHABGECFkajcAPgAGHOHAD0AKjjARktFiRxnjjjaccDx8AgBCAK4wMHzintLYMkZRMRR2DuoIwnz4CPYOcLzo0IZQOEFwuAGcHMqR0bGa2rqYYggwUBGY1gDC6O7McCCYyo71qVZAA)

**Comment by Mad-Kat** [comment]
After @rvetere showed me the whole code, I identified that we have an issue when the component isn't a native component, but instead wraps another component.

[Playground](https://www.typescriptlang.org/play/?jsx=4#code/JYWwDg9gTgLgBAbwM4wJ4BsCmATAvnAMyghDgCIA7TADxgFpUBDAazIG4AoDgYwgpTgB1KIzBwAvHBQYcAOgBGAVxgw+AAzWcefAQEleFCVLRZsspADcA5gB4EcACSNuMYBcwAuOPIgQsjQ1wAPjUOODgAegAqOGADYxkkOCiIjk0w7X54ACFlVUNJaVMACmFRAEpQ8IcEfT58BDDwuF50aC8oHE5w3DStCIi4AAEYJDoaMEwXcahiKEdnV3cpAAsIRXRsOAoIeHlMOE6AR0VgTuwOG1yVPgWXN0xxBBgoRUxggGF0OOY4EEwbBFrvkgkA)

**Comment by jantimon** [comment]
@rvetere released as next-yak 1.2.1
